opentelemetry/proto/profiles/v1development/eventprofile.proto

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opentelemetry.proto.profiles.v1development;
+
+import "opentelemetry/proto/common/v1/common.proto";
+
+// Profile holds a collection of events that represent a profile.
+message Profile {
```

Comment:
Did you consider having a single `event_type` for a `message Profile`?
For `message Profile` with a single `event_type` processing could look like this:
```mermaid
flowchart TD
    A[Shared Envelope] --> B{" "}
    B --> D[Profile for on-CPU profiling]
    B --> E[Profile for off-CPU profiling]
    B --> F[Profile for memory allocation profiling]
```

The string table, attribute table and other shared data then should be part of the `Shared Envelope` (I'm bad in naming things :pray: ). This would allow faster handling of profiling data by processors based on `event_type`. With the suggested approach processors have to walk all `events` to handle a specific `event_type`.

Reply:
That's an interesting idea. I can see how this could simplify things. I'll think about this a bit more!